### PR TITLE
feat: OrgUnitUserGroup OCC and Ngrx

### PR DESCRIPTION
**Pull request by Zeyber**
Closes: #6537

Actions Tested: 
- [x] [OrgUnitUserGroup] Load OrgUnitUserGroup Data
- [x] [OrgUnitUserGroup] Load OrgUnitUserGroup Data Fail
- [x] [OrgUnitUserGroup] Load OrgUnitUserGroup Data Success
- [x] [OrgUnitUserGroup] Load OrgUnitUserGroups
- [x] [OrgUnitUserGroup] Load OrgUnitUserGroups Fail
- [x] [OrgUnitUserGroup] Load OrgUnitUserGroups Success
- [x] [OrgUnitUserGroup] Create OrgUnitUserGroup
- [x] [OrgUnitUserGroup] Create OrgUnitUserGroup Fail
- [x] [OrgUnitUserGroup] Create OrgUnitUserGroup Success
- [x] [OrgUnitUserGroup] Update OrgUnitUserGroup
- [x] [OrgUnitUserGroup] Update OrgUnitUserGroup Fail
- [x] [OrgUnitUserGroup] Update OrgUnitUserGroup Success
- [x] [OrgUnitUserGroup] Delete OrgUnitUserGroup
- [x] [OrgUnitUserGroup] Delete OrgUnitUserGroup Fail
- [x] [OrgUnitUserGroup] Delete OrgUnitUserGroup Success

### CODE REVIEW

File: projects/core/src/occ/adapters/organization/occ-org-unit-user-group.adapter.ts [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+  ): Observable<EntitiesModel<OrgUnitUserGroup>> {
+    return this.http
+      .get<Occ.OrgUnitUserGroupList>(
+        this.getOrgUnitUserGroupAvailableOrderApprovalPermissionsEndpoint(
+          userId,
+          orgUnitUserGroupUid,
+          params
+        )
+      )
+      .pipe(this.converter.pipeable(ORG_UNIT_USER_GROUPS_NORMALIZER));
```
Comment (Pucek9) [inline projects/core/src/occ/adapters/organization/occ-org-unit-user-group.adapter.ts]:
```suggestion
      .pipe(this.converter.pipeable(PERMISSIONS_NORMALIZER));
```

File: projects/core/src/occ/adapters/organization/occ-org-unit-user-group.adapter.ts [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+  ): Observable<EntitiesModel<User>> {
+    return this.http
+      .get<Occ.OrgUnitUserGroupList>(
+        this.getOrgUnitUserGroupAvailableOrgCustomersEndpoint(
+          userId,
+          orgUnitUserGroupUid,
+          params
+        )
+      )
+      .pipe(this.converter.pipeable(ORG_UNIT_USER_GROUPS_NORMALIZER));
```
Comment (Pucek9) [inline projects/core/src/occ/adapters/organization/occ-org-unit-user-group.adapter.ts]: I know that's not created yet, but for remember
```suggestion
      .pipe(this.converter.pipeable(ORG_UNIT_USERS_NORMALIZER));
```

File: projects/core/src/occ/adapters/organization/default-occ-organization-config.ts [hunk 3]
```diff
@@ -12,6 +12,21 @@ export const defaultOccOrganizationConfig: OccConfig = {
           '/users/${userId}/orgUnitsAvailableApprovalProcesses',
         orgUnits: '/users/${userId}/orgUnits',
         orgUnit: '/users/${userId}/orgUnits/${orgUnitId}',
+        orgUnitUserGroups: '/users/${userId}/orgUnitUserGroups',
```
Comment (Pucek9) [inline projects/core/src/occ/adapters/organization/default-occ-organization-config.ts]: I think `userGroup` is enougth here as preffix

File: projects/core/src/occ/adapters/organization/default-occ-organization-config.ts [hunk 4]
```diff
@@ -12,6 +12,21 @@ export const defaultOccOrganizationConfig: OccConfig = {
...
+        orgUnitUserGroupAvailableOrgCustomers:
+          '/users/${userId}/orgUnitUserGroups/${orgUnitUserGroupUid}/availableOrgCustomers',
+        orgUnitUserGroupMembers:
+          '/users/${userId}/orgUnitUserGroups/${orgUnitUserGroupUid}/members',
+        orgUnitUserGroupMember:
+          '/users/${userId}/orgUnitUserGroups/${orgUnitUserGroupUid}/members/${orgCustomerId}',
+        orgUnitUserGroupOrderApprovalPermissions:
+          '/users/${userId}/orgUnitUserGroups/${orgUnitUserGroupUid}/orderApprovalPermissions',
+        orgUnitUserGroupOrderApprovalPermission:
+          '/users/${userId}/orgUnitUserGroups/${orgUnitUserGroupUid}/orderApprovalPermissions/${orderApprovalPermissionCode}',
```
Comment (Pucek9) [inline projects/core/src/occ/adapters/organization/default-occ-organization-config.ts:29]: I think `userGroup` is enougth here as preffix

Comment (Zeyber) [reply projects/core/src/occ/adapters/organization/default-occ-organization-config.ts:29]: I'd like to keep the full name to match the API call. It keeps it distinguished and provides the flexibility to add different types of "user groups" in the future.

**Review by Pucek9 — APPROVED** [review]

**Pull request merged**